### PR TITLE
Export types

### PR DESCRIPTION
**Pull request by wagenet**

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -17,12 +17,16 @@ import {
 } from './backburner/utils';
 
 import searchTimer from './backburner/binary-search';
+
 import DeferredActionQueues from './backburner/deferred-action-queues';
+export { DeferredActionQueues };
```
Comment (rwjblue) [inline lib/index.ts:22]: Should this be `export type { DeferredActionQueues }`? I don't think we actually want to export the value, just the type...

File: package.json [hunk 2]
```diff
@@ -21,22 +22,25 @@
...
-    "tslint": "^5.11.0"
+    "lolex": "^6.0.0",
+    "qunit": "^2.14.1",
+    "rollup-plugin-buble": "^0.19.8",
+    "rollup-plugin-commonjs": "^10.1.0",
+    "rollup-plugin-node-resolve": "^5.2.0",
+    "tslint": "^6.1.3"
+  },
+  "engines": {
+    "node": "10.* || >= 12.*"
```
Comment (rwjblue) [inline package.json:44]: Do we need this? Adding it is basically a breaking change, but none of the other changes seem breaking...

File: tests/queue-test.ts [hunk 3]
```diff
@@ -73,10 +73,10 @@ QUnit.test('Default queue can be manually configured', function(assert) {
...
   assert.expect(2);
 
-  let befores: Array<any | null | undefined> = [];
-  let afters: Array<any | null | undefined> = [];
-  let expectedBefores: Array<any | null | undefined> = [];
-  let expectedAfters: Array<any | null | undefined> = [];
+  let befores: (any | null | undefined)[] = [];
+  let afters: (any | null | undefined)[] = [];
+  let expectedBefores: (any | null | undefined)[] = [];
+  let expectedAfters: (any | null | undefined)[] = [];
```
Comment (rwjblue) [inline tests/queue-test.ts:79]: 1) seems like the same thing, is htis just preference?
2) why didn't we use `unknown[]`?

File: tsconfig.json [hunk 4]
```diff
@@ -2,12 +2,13 @@
   "compilerOptions": {
     "moduleResolution": "node",
     "target": "es2015",
-    "noEmit": true,
+    // "noEmit": true,
```
Comment (rwjblue) [inline tsconfig.json:5]: comments in json make me sad, if we don't want this option anymore lets just delete it

File: package.json (hunk 2 above)
Comment (wagenet) [reply package.json:44]: I was having trouble running things locally without it, but I can try again.

File: tests/queue-test.ts (hunk 3 above)
Comment (wagenet) [reply tests/queue-test.ts:79]: `tslint` wanted it to change. No clue why.

File: tsconfig.json (hunk 4 above)
Comment (wagenet) [reply tsconfig.json:5]: Good call. I'll check

File: lib/index.ts (hunk 1 above)
Comment (wagenet) [reply lib/index.ts:22]: Probably.

File: tsconfig.json (hunk 4 above)
Comment (wagenet) [reply tsconfig.json]: Unnecessary. I removed it.

File: package.json (hunk 2 above)
Comment (wagenet) [reply package.json]: Looks like it was unnecessary. I removed it.

**Review by rwjblue — COMMENTED** [review]
Ready for a rebase now...

File: ember-cli-build.js [hunk 5]
```diff
@@ -9,6 +9,7 @@ const path = require('path');
 const typescript = require('broccoli-typescript-compiler').default;
 const buble = require('rollup-plugin-buble');
 const fs = require('fs');
+const { relative } = require('path');
```
Comment (rwjblue) [inline ember-cli-build.js:12]: I _think_ this is unused?
```suggestion
```

**Review by rwjblue — APPROVED** [review]

**Pull request merged**